lib/MetagenomeAPI/MetagenomeAPIImpl.py

```diff
@@ -252,6 +272,85 @@ def search_region(self, ctx, params):
+           values, false = 0, true = 1 @range [0,1])
+        :returns: instance of type "SearchContigsResult" -> structure:
+           parameter "num_found" of Long, parameter "start" of Long,
+           parameter "contigs" of list of type "contig" -> structure:
+           parameter "contig_id" of String, parameter "feature_count" of
+           Long, parameter "length" of Long
+        """
+        # ctx is the context object
+        # return variables are: result
+        #BEGIN search_contigs
+        if 'ref' not in params:
+          raise RuntimeError(f"'ref' argument required for search_contigs")
+        if 'limit' not in params:
+          raise RuntimeError(f"'limit' argument required for search_contigs")
+        if 'start' not in params:
+          raise RuntimeError(f"'start' argument required for search_contigs")
+        if not params.get('sort_by'):
+          sort_by = ('length', 0)
+        else:
+          sort_by = params['sort_by']
+          if sort_by[0] not in ['length', 'contig_id', 'feature_count']:
+            raise RuntimeError("'sort_by' argument can only contain one "
+                               "of 'length', 'contig_id', or 'feature_count'")
+
+        ws = Workspace(self.config['workspace-url'], token=ctx['token'])
+        ama_utils = AMAUtils(ws)
+        params['included_fields'] = ['contig_ids', 'contig_lengths']
+        ama = ama_utils.get_annotated_metagenome_assembly(params)['genomes'][0]
+        data = ama['data']
+        if len(data['contig_ids']) != len(data['contig_lengths']):
+          raise RuntimeError(f"contig ids (size: {len(data['contig_ids'])}) and contig "
+                             f"lengths (size: {len(data['contig_lengths'])}) sizes do not match.")
+        contig_ids = data['contig_ids']
+        contig_lengths = data['contig_lengths']
+
+        if sort_by[0] == 'contig_id' and not sort_by[1]:
+          contig_ids, contig_lengths = (list(t) for t in zip(*sorted(zip(contig_ids, contig_lengths), reverse=True)))
+        elif sort_by[0] == 'length':
+          contig_lengths, contig_ids = (list(t) for t in zip(*sorted(zip(contig_lengths, contig_ids), reverse=sort_by[1] == 0)))
+
+        contigs = []
+        for i in range(params['start'], params['start'] + params['limit']):
+          # not a great solution, but works for now.
+          if i < len(contig_ids) or i >= 0:
+            contig_id = contig_ids[i]
+            contig_length = contig_lengths[i]
+            contig_data = self.msu.search_contig(ctx["token"],
+                                          params.get("ref"),
+                                          contig_id,
+                                          contig_length,
+                                          params.get("start"),
+                                          params.get("limit"))
+            contigs.append(contig_data)
+        # not ideal, but will work for now...
+        if sort_by[0] == 'feature_count':
```

Comment:
This will only sort your first page of results and not all results, right? Maybe this will be confusing to the user if they expect the sort to be applied to all results?

Reply:
yeah, the issue here is that I would have to find all the 'feature_count' fields by querying the searchapi2 for each  contig. My gut feeling is that it would be very slow